client/landing/jetpack-cloud/components/date-picker/index.jsx

```diff
@@ -18,80 +19,107 @@ import Gridicon from 'components/gridicon';
+	getDisplayDate = ( date, showTodayYesterday = true ) => {
+		const { moment, translate } = this.props;
+
+		const daysDiff = moment().diff( date, 'days' );
+		const yearToday = moment().format( 'YYYY' );
+		const yearDate = moment( date ).format( 'YYYY' );
+
+		const dateFormat = yearToday === yearDate ? 'MMM D' : 'MMM D, YYYY';
 
-	getDisplayDate = dateString => {
-		const word = this.props.moment
-			.parseZone( dateString )
-			.calendar()
-			.split( ' ' )[ 0 ];
-		if ( 'Today' === word || 'Yesterday' === word ) {
-			return word;
+		if ( showTodayYesterday ) {
+			switch ( daysDiff ) {
+				case 0:
+					return translate( 'Today' );
+				case 1:
+					return translate( 'Yesterday' );
+			}
 		}
 
-		return this.getFormattedDate( dateString );
+		return moment( date ).format( dateFormat );
 	};
 
 	shuttleLeft = () => {
-		const { moment, onChange, selectedDateString } = this.props;
-		const newDateString = moment
-			.parseZone( selectedDateString )
-			.subtract( 1, 'days' )
-			.toISOString( true );
+		if ( ! this.canShuttleLeft() ) {
+			return false;
+		}
+
+		const { moment, onDateChange, selectedDate } = this.props;
 
-		onChange( newDateString );
+		const newSelectedDate = moment( selectedDate ).subtract( 1, 'days' );
+
+		onDateChange( newSelectedDate.toDate() );
 	};
 
 	shuttleRight = () => {
 		if ( ! this.canShuttleRight() ) {
 			return false;
 		}
 
-		const { moment, onChange, selectedDateString } = this.props;
-		const newDateString = moment
-			.parseZone( selectedDateString )
-			.add( 1, 'days' )
-			.toISOString( true );
+		const { moment, onDateChange, selectedDate } = this.props;
+
+		const newSelectedDate = moment( selectedDate ).add( 1, 'days' );
+
+		onDateChange( newSelectedDate.toDate() );
+	};
+
+	canShuttleLeft = () => {
+		const { moment, selectedDate, oldestDateAvailable } = this.props;
 
-		onChange( newDateString );
+		return !! oldestDateAvailable && ! moment( selectedDate ).isSame( oldestDateAvailable, 'day' );
 	};
 
 	canShuttleRight = () => {
-		const { moment, selectedDateString } = this.props;
-		return ! moment().isSame( moment.parseZone( selectedDateString ), 'day' );
+		const { moment, selectedDate } = this.props;
+
+		return ! moment( selectedDate ).isSame( moment(), 'day' );
 	};
 
 	render() {
-		const { selectedDateString, siteId } = this.props;
+		const { selectedDate, siteId, moment } = this.props;
+
+		const previousDate = moment( selectedDate ).subtract( 1, 'days' );
+		const nextDate = moment( selectedDate ).add( 1, 'days' );
 
-		const currentDisplayDate = this.getDisplayDate( selectedDateString );
+		const previousDisplayDate = this.getDisplayDate( previousDate );
+		const nextDisplayDate = this.getDisplayDate( nextDate, false );
 
 		return (
 			<div className="date-picker">
 				<Button compact borderless onClick={ this.shuttleLeft }>
-					<Gridicon icon="chevron-left" />
+					<Gridicon icon="chevron-left" className={ ! this.canShuttleLeft() && 'disabled' } />
```

Comment:
Why even show this at all? instead of disabeling it?

Reply:
That is like the design. I prefer the option of hiding it. We could discuss it with the design team.